ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/-create-run-utils.ts

```diff
@@ -0,0 +1,464 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { z } from 'zod';
+
+import { CreateOrtRun, OrtRun } from '@/api/requests';
+import { packageManagers } from './-types';
+
+const keyValueSchema = z.object({
+  key: z.string().min(1),
+  value: z.string(), // Allow empty values for now
+});
+
+export const createRunFormSchema = z.object({
+  revision: z.string(),
+  path: z.string(),
+  jobConfigs: z.object({
+    analyzer: z.object({
+      enabled: z.boolean(),
+      repositoryConfigPath: z.string().optional(),
+      allowDynamicVersions: z.boolean(),
+      skipExcluded: z.boolean(),
+      enabledPackageManagers: z.array(z.string()),
+    }),
+    advisor: z.object({
+      enabled: z.boolean(),
+      skipExcluded: z.boolean(),
+      advisors: z.array(z.string()),
+    }),
+    scanner: z.object({
+      enabled: z.boolean(),
+      skipConcluded: z.boolean(),
+      skipExcluded: z.boolean(),
+    }),
+    evaluator: z.object({
+      enabled: z.boolean(),
+      ruleSet: z.string().optional(),
+      licenseClassificationsFile: z.string().optional(),
+      copyrightGarbageFile: z.string().optional(),
+      resolutionsFile: z.string().optional(),
+    }),
+    reporter: z.object({
+      enabled: z.boolean(),
+      formats: z.array(z.string()),
+    }),
+    notifier: z.object({
+      enabled: z.boolean(),
+      notifierRules: z.string().optional(),
+      resolutionsFile: z.string().optional(),
+      mail: z.object({
+        recipientAddresses: z.array(z.object({ email: z.string() })).optional(),
+        mailServerConfiguration: z.object({
+          hostName: z.string(),
+          port: z.coerce.number().int(),
+          username: z.string(),
+          password: z.string(),
+          useSsl: z.boolean(),
+          fromAddress: z.string(),
+        }),
+      }),
+      jira: z.object({
+        jiraRestClientConfiguration: z.object({
+          serverUrl: z.string(),
+          username: z.string(),
+          password: z.string(),
+        }),
+      }),
+    }),
+    parameters: z.array(keyValueSchema).optional(),
+  }),
+  labels: z.array(keyValueSchema).optional(),
+  jobConfigContext: z.string().optional(),
+});
+
+export type CreateRunFormValues = z.infer<typeof createRunFormSchema>;
+
+/**
+ * Get the default values for the create run form. The form can be provided with a previously run
+ * ORT run, in which case the values from it are used as defaults. Otherwise uses base defaults.
+ */
+export function defaultValues(
+  ortRun: OrtRun | null
+): z.infer<typeof createRunFormSchema> {
+  // Default values for the form: edit only these, not the defaultValues object.
+  const baseDefaults = {
+    revision: 'main',
+    path: '',
+    jobConfigs: {
+      analyzer: {
+        enabled: true,
+        repositoryConfigPath: '',
```

Comment:
Was the verdict that we should have `undefined` here?

Reply:
Probably yes. This PR only moves the existing logic to functions, so it's easier to work on and possibly write tests for.